akka-docs/src/main/paradox/stream/stream-rate.md

```diff
@@ -203,28 +203,48 @@ Scala
 Java
 :   @@snip [RateTransformationDocTest.java]($code$/java/jdocs/stream/RateTransformationDocTest.java) { #conflate-sample }
 
-### Understanding expand
+### Understanding extrapolate and expand
 
-Expand helps to deal with slow producers which are unable to keep up with the demand coming from consumers.
-Expand allows to extrapolate a value to be sent as an element to a consumer.
+Now we will discuss two stages, `extrapolate` and `expand`, helping to deal with slow producers that are unable to keep up with the demand coming from consumers.
+They allow for additional values to be sent as elements to a consumer.
 
-As a simple use of `expand` here is a flow that sends the same element to consumer when producer does not send
-any new elements.
+As a simple use case of `extrapolate`, here is a flow that repeats the last emitted element to a consumer, whenever the consumer signals demand and the producer cannot supply new elements yet.
 
 Scala
-:   @@snip [RateTransformationDocSpec.scala]($code$/scala/docs/stream/RateTransformationDocSpec.scala) { #expand-last }
+:   @@snip [RateTransformationDocSpec.scala]($code$/scala/docs/stream/RateTransformationDocSpec.scala) { #extrapolate-last }
 
 Java
-:   @@snip [RateTransformationDocTest.java]($code$/java/jdocs/stream/RateTransformationDocTest.java) { #expand-last }
+:   @@snip [RateTransformationDocTest.java]($code$/java/jdocs/stream/RateTransformationDocTest.java) { #extrapolate-last }
 
-Expand also allows to keep some state between demand requests from the downstream. Leveraging this, here is a flow
-that tracks and reports a drift between fast consumer and slow producer.
+For situations where you have several producers, and the entire stream may become blocked by a non-emitting one, you can use the `initial` parameter of `extrapolate` to "seed" the stream.
+
+Scala
+:   @@snip [RateTransformationDocSpec.scala]($code$/scala/docs/stream/RateTransformationDocSpec.scala) { #extrapolate-seed }
+
+Java
+:   @@snip [RateTransformationDocTest.java]($code$/java/jdocs/stream/RateTransformationDocTest.java) { #extrapolate-seed }
+
+`extrapolate` and `expand` also allow to produce metainformation based on demand signalled from the downstream. Leveraging this, here is a flow
+that tracks and reports a drift between a fast consumer and a slow producer. 
+
+Scala
+:   @@snip [RateTransformationDocSpec.scala]($code$/scala/docs/stream/RateTransformationDocSpec.scala) { #extrapolate-drift }
+
+Java
+:   @@snip [RateTransformationDocTest.java]($code$/java/jdocs/stream/RateTransformationDocTest.java) { #extrapolate-drift }
+
+And here's a more concise representation with `expand`.
```

Comment:
I think here you should put a more detailed description of the difference between `expand` and `extrapolate`, something like this:

"While `extrapolate` is only applied when there is back pressure, `expand` will always applied to the element going through the stream, this makes some additional things such as transformations and filtering possible.

Here is the previous example with `expand` which makes it somewhat more concise:" 

And then maybe just skip the text below

Reply:
I'd prefer to leave the more verbose explanation where it is - from my observations readers often "zone out" to the explanations in-between when reading snippets, having it at the end will serve to reduce that effect.

I've reworded the subsequent text to refer to the examples more directly.